include/swift/AST/ModuleDependencies.h

```diff
@@ -606,15 +642,15 @@ class SwiftDependencyScanningService {
   ///
   /// \returns the cached result, or \c None if there is no cached entry.
   Optional<ModuleDependencyInfo>
-  findDependencies(StringRef moduleName,
-                   Optional<ModuleDependencyKind> kind) const;
+  findDependency(StringRef moduleName,
+                 Optional<ModuleDependencyKind> kind) const;
 
   /// Record dependencies for the given module.
-  const ModuleDependencyInfo *recordDependencies(StringRef moduleName,
+  const ModuleDependencyInfo *recordDependency(StringRef moduleName,
```

Comment:
One thing I'm not quite following is the de-pluralization of these method names. Aren't there potentially multiple dependencies of `moduleID` in the given `ModuleDependencyInfo` collection?

Reply:
While there are, indeed, multiple dependencies of `moduleID` in the given `ModuleDependencyInfo`, the thing being recorded here is the existence of a `ModuleDependencyInfo` itself for this `moduleID`. Which does contain a collection of its own _dependencies_' IDs, each of which will then correspond to a `ModuleDependencyInfo` to be queried via these same API, but also a bunch of other metadata we need to know about this _dependency_.

I can see the logic behind the pluralization but it's been bugging me for a while and it seems to be clearer to be explicit about the fact that these API are vending these info values that describe a particular module, not just that module's dependencies. 

For example, before this patch `hasDependencies("foo")` would indicate whether or not a module `"foo"` has been discovered as a _dependency_ during this scan. To your point, it does also indicate that we know _dependencies_ of `foo` in its respective info, but the use of these queries is broader than querying dependencies of this module as it also covers getting a dependency info about this module for other purposes as well. It seems conceptually more-accurate to have some thing like `findDependencies("foo")` return an actual list of dependencies of `foo`, rather than an info describing `foo` as a discovered dependency itself. 

I'm having to really make an argument here so I can see it either way and can be convinced otherwise, but this seemed more appropriate as I was already restructuring things!